# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/wmf/WMFDecoderModule.cpp
# Commit: d158b20d9a65
# Full Hash: d158b20d9a65cd05dd69a586473216b800523083
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - PD: Allow WMF decoding inside RDD process. r=mattwoodrow
#   
#   This assumes that access to Windows WMF and DirectX is allowed from RDD process.
#   
#   Depends on D91699
# ==============================================================================

diff -r d0c9c64867cc -r d158b20d9a65 dom/media/platforms/wmf/WMFDecoderModule.cpp
--- a/dom/media/platforms/wmf/WMFDecoderModule.cpp	Tue Oct 20 23:26:49 2020 +0000
+++ b/dom/media/platforms/wmf/WMFDecoderModule.cpp	Tue Oct 20 23:26:52 2020 +0000
@@ -104,8 +104,8 @@
     // directly calls WMFDecoderModule::Supports in the content process.
     // This unnecessary requirement will be fixed in bug 1534815.
     testForVPx = true;
-  } else if (XRE_IsGPUProcess()) {
-    // Always allow DXVA in the GPU process.
+  } else if (XRE_IsGPUProcess() || XRE_IsRDDProcess()) {
+    // Always allow DXVA in the GPU or RDD process.
     testForVPx = sDXVAEnabled = true;
   } else {
     // Only allow DXVA in the UI process if we aren't in e10s Firefox